novawallet/Common/Services/ExtrinsicService/Substrate/FeeManaging/ExtrinsicFeeEstimationRegistry.swift

```diff
@@ -91,39 +122,88 @@ extension ExtrinsicFeeEstimationRegistry: ExtrinsicFeeEstimationRegistring {
             }
 
             return createFeeInstallerWrapper(
-                chain: chain,
-                asset: asset,
-                chainAssetId: chainAssetId
+                senderResolutionOperation: senderResolutionOperation,
+                chainAsset: ChainAsset(chain: chain, asset: asset)
             )
         } catch {
             return CompoundOperationWrapper.createWithError(error)
         }
     }
 
     func createFeeInstallerWrapper(
-        chain: ChainModel,
-        asset: AssetModel,
-        chainAssetId: ChainAssetId
+        senderResolutionOperation: ClosureOperation<ExtrinsicSenderBuilderResolution>,
+        chainAsset: ChainAsset
     ) -> CompoundOperationWrapper<ExtrinsicFeeInstalling> {
-        switch AssetType(rawType: asset.type) {
+        switch AssetType(rawType: chainAsset.asset.type) {
         case .none:
             CompoundOperationWrapper.createWithResult(ExtrinsicNativeFeeInstaller())
         case .statemine where chain.hasAssetHubTransferFees:
             CompoundOperationWrapper.createWithResult(
                 ExtrinsicAssetConversionFeeInstaller(
-                    feeAsset: ChainAsset(chain: chain, asset: asset)
+                    feeAsset: chainAsset
                 )
             )
         case .orml where chain.hasHydrationTransferFees:
-            CompoundOperationWrapper.createWithResult(
-                HydraExtrinsicFeeInstaller(
-                    feeAsset: ChainAsset(chain: chain, asset: asset)
-                )
+            createHydraFeeInstallingWrapper(
+                senderResolutionOperation: senderResolutionOperation,
+                chainAsset: chainAsset
             )
         case .orml, .statemine, .equilibrium, .evmNative, .evmAsset:
             .createWithError(
-                ExtrinsicFeeEstimationRegistryError.unexpectedChainAssetId(chainAssetId)
+                ExtrinsicFeeEstimationRegistryError.unexpectedChainAssetId(chainAsset.chainAssetId)
+            )
+        }
+    }
+
+    private func createHydraFeeInstallingWrapper(
+        senderResolutionOperation: ClosureOperation<ExtrinsicSenderBuilderResolution>,
+        chainAsset: ChainAsset
+    ) -> CompoundOperationWrapper<ExtrinsicFeeInstalling> {
+        let swapStateWrapper = OperationCombiningService.compoundNonOptionalWrapper(
+            operationManager: OperationManager(operationQueue: OperationQueue())
```

Comment:
OperationQueue must come from init together with ChainRegistry and StorageFacades